providers/azure.go

```diff
@@ -149,13 +145,103 @@ func (p *AzureProvider) Redeem(ctx context.Context, redirectURL, code string) (*
 	created := time.Now()
 	expires := time.Unix(jsonResponse.ExpiresOn, 0)
 
-	return &sessions.SessionState{
+	session := &sessions.SessionState{
 		AccessToken:  jsonResponse.AccessToken,
 		IDToken:      jsonResponse.IDToken,
 		CreatedAt:    &created,
 		ExpiresOn:    &expires,
 		RefreshToken: jsonResponse.RefreshToken,
-	}, nil
+	}
+
+	email, err := p.verifyTokensAndExtractEmail(ctx, session.IDToken, session.AccessToken)
+	if err != nil {
+		return nil, err
```

Comment:
Do you really want to completely fail and return a nil session if email claims to session extraction failed? Isn't the `EnrichSession` call after this hitting the Profile URL the fallback to get an email via that mechanism?

Reply:
right, I will update